aip/general/0131.md

```diff
@@ -21,7 +21,9 @@ resource.
 ## Guidance
 
 APIs **must** provide a get method for resources. The purpose of the get method
-is to return data from a single resource.
+is to return data from a single resource. If the data being retrieved is not an
+([AIP-122][]) compliant resource, please see ([AIP-136][]) to add a Custom Method
```

Comment:
AIP-131 is specifically targeted at resource-oriented, AIP-122 compliant APIs. This statement is redundant IMO, why do you believe readers of these AIPs would benefit from this clarification?

Reply:
Personally, I think it's just a nice way to be explicit for users that are working on creating APIs for "other resources" - since "resource" itself is quite an overused term. And users might start here anyway to take a look at the usual Get documentation, so a quick reminder that custom methods should be used for those "other resources" seems like it could be helpful.

Reply:
If the intent of this change is to guide API producers on how to write AIP-compliant API surfaces, having recommendations for non-aip compliant resources goes against that. In general we must not add recommended paths for writing APIs that do not comply with AIPs.

If there's confusion around what the definition of a resource is, we should clarify that in AIP-121.

Reply:
> having recommendations for non-aip compliant resources goes against that

I don't disagree, I think it can be confusing for API reviewers when changes come in for non-aip compliant resources that look like they are compliant resources. Here's the CL that had got me thinking about this change: https://critique.corp.google.com/cl/615099995

That said, if you still think we are better off without this, I'll revert the change

Reply:
_tl;dr: I'm preferring explicit "do X, or do Y" guidance these days, rather than depending on readers to saturate their brains with all context of related AIPs. I lean towards @Berg3's suggestion as a result. But maybe we can dial in the linter..._

I think this rule (and class of rule for other standard methods) has the same problem as https://linter.aip.dev/123/resource-annotation where we want to nudge users towards a common pattern for X examples, and _away_ from that common pattern for inverse-X examples, by trying to intuit their intentions in the proto and often get it wrong. Put in context, we want Standard Get methods to be named/structured as `Get{Resource}`, and we want things that aren't Standard Get to be named/structured as something else. The problem then becomes that the author either A. takes the guidance at face value or B. disables the rule, and both have a lot of potential to be incorrect actions if the ApiLinter incorrectly warned on the element.

Tightening the heuristic has the problem of potentially missing out on those in between scenarios e.g., "Did you mean `rpc GetBook(GetBookRequest) returns (Book)` instead of `rpc FetchBook(FetchBookRequest) returns (FetchBookResponse)`???".

Maybe we need to be looking at if the response is/contains a resource?

Reply:
Hey - going to revert this change, thanks for the feedback!